model/tox.ini

```diff
@@ -9,14 +9,17 @@ skipsdist = true
 
 [testenv]
 passenv =
+    CUDA_PATH
+    CUDAARCHS
+    NUM_PROCESSES
     PIP_USER
     PYTHONUSERBASE
     TEST_DATA_PATH
 deps =
     -r {toxinidir}/requirements-dev.txt
 commands =
     pip list
-    -pytest -v -s -n auto --cache-clear --cov --cov-reset --doctest-modules atmosphere/dycore/src atmosphere/diffusion/src atmosphere/advection/src common/src driver/src
+    -pytest -v -s -n {env:NUM_PROCESSES:1} --cache-clear --cov --cov-reset --doctest-modules atmosphere/dycore/src atmosphere/diffusion/src atmosphere/advection/src common/src driver/src
```

Comment:
Is there a particular reason why you limit the num processes to 16 here?

Reply:
Yes. If we leave `auto` pytest will spawn 244 threads on the gh200 cpu, one for each core. In gt4py we have observed 2 kinds of issues with high parallelism: each test is running cmake and building a program from source files, and many builds in parallel could cause the error that too many files are open at the same time; it seems that the cuda MPS server was hanging sometimes. So I apply in icon4py the same restriction we applied in gt4py.